src/bootstrap/setup.rs

```diff
@@ -107,6 +108,19 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let include_path = profile.include_path(src_path);
     println!("`x.py` will now use the configuration at {}", include_path.display());
 
+    let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
+    let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
+    let toolchain_link =
+        Command::new("rustup").args(&["toolchain", "link", "stage1", &stage_path[..]]).output();
```

Comment:
I just realized this will overwrite any existing toolchain - that seems kind of not great, it means we'll delete any existing toolchain with this name. @kinnison @rbtcollins is that intentional that it will silently overwrite?

I guess to avoid that we could explicitly run `rustup toolchain list` and see if it already has `stage1`.

Reply:
Usually this command is executed by the user, so like most Unix-y commands, dispatches as silently and swiftly as possible, and alerting someone is a sign of failure.

Reply:
Hmm - but `ln -s` (which you said in the original issue is basically what `toolchain link` does) gives an error if the file already exists:
```
$ ln -s x config.toml
ln: failed to create symbolic link 'config.toml': File exists
```

Reply:
Reasonable~
But the other reason I mentioned links is due to problems due to OS control over when files may be edited/clobbered/removed, which has less to do with the particular link.

Reply:
Since we have to create the directory in the first place, couldn't we just not link if it already exists?

Reply:
> Since we have to create the directory in the first place, couldn't we just not link if it already exists?

That doesn't seem great - that won't warn or setup the toolchain if the user has already run `x.py build` but not created a rustup toolchain.

Reply:
I think this is why @Mark-Simulacrum suggested using a stamp file. We could also extend this by adding configuration for it in the `config.toml`; e.g. `toolchain-overwrite` or something where the user can specify if they want to overwrite the toolchain every time they run `x.py setup` or not, and by default `./x.py setup` will only do it once.

Reply:
I really don't think this needs additional configuration. We can just check if `rustup toolchain list` contains `stage1`.